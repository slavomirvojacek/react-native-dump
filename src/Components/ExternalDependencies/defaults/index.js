import { noop } from '@openmaths/utils'

const props = {
  logIntoConsole: noop,
}

export {
  props
}
