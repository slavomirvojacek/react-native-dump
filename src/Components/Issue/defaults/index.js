const issueDefault = (override = {}) => {
  const def = {
    id: '',
    title: '',
    url: '',
    bodyText: '',
    reactions: {
      nodes: []
    }
  }

  return {...def, ...override}
}

const props = {
  issue: issueDefault(),
}

export {
  issueDefault, props
}
