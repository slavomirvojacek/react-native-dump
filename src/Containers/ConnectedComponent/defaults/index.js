import { None } from '@threestup/monads'

const props = {
  showLoading: false,
  username: None,
  error: None,

}

export {
  props
}
