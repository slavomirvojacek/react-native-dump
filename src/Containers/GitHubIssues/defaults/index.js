const userDefault = (id = '') => ({id})

const reactionDefault = (override = {}) => {
  const def = {
    id: '',
    content: '',
    user: userDefault()
  }

  return {...def, ...override}
}

const issueDefault = (override = {}) => {
  const def = {
    id: '',
    title: '',
    url: '',
    bodyText: '',
    reactions: []
  }

  return {...def, ...override}
}

export {
  userDefault, reactionDefault, issueDefault
}
