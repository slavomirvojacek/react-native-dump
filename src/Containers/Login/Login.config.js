import PropTypes from 'prop-types'

import { initialState } from '../../Redux/Login/reducer'

const displayName = 'LoginContainer'

const propTypes = {
  loginState: PropTypes.shape({
    data: PropTypes.object.isRequired,
    isPending: PropTypes.bool.isRequired,
    error: PropTypes.object.isRequired
  }),
  login: PropTypes.func.isRequired
}

const defaultProps = {
  loginState: initialState,
  login: username => null
}

export default {
  displayName, propTypes, defaultProps
}
