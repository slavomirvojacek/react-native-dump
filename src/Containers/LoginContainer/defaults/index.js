import { noop } from '@openmaths/utils'

import { initialState } from '../lib/reducer'

const props = {
  signIn: noop,
  loginState: initialState,
}

export {
  props
}
