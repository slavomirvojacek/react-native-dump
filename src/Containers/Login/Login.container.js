import React, { Component } from 'react'
import { View, Text } from 'react-native'
import { connect } from 'react-redux'

import Config from './Login.config'
import styles from './Login.container.styles'
import { apiFetch } from '../../Redux/Login/saga'
import LoginForm from '../../Forms/Login/Login.form'

class LoginContainer extends Component {
  onSignIn = values => {
    this.props.login(values)
  }

  render () {
    return (
      <View style={styles.container}>
        <LoginForm style={styles.container} onSubmit={this.onSignIn}/>

        <Text style={styles.container}>
          {JSON.stringify(this.props.loginState)}
        </Text>
      </View>
    )
  }
}

LoginContainer.displayName = Config.displayName
LoginContainer.propTypes = Config.propTypes
LoginContainer.defaultProps = Config.defaultProps

export const mapStateToProps = ({Login}) => ({
  loginState: Login
})

const mapDispatchToProps = dispatch => ({
  login: username => dispatch(apiFetch(username))
})

export { LoginContainer }
export default connect(mapStateToProps, mapDispatchToProps)(LoginContainer)
