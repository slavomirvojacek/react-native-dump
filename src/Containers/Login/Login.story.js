import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { action } from '@storybook/addon-actions'

import { LoginContainer } from './Login.container'
import { initialState as LoginState } from '../../Redux/Login/reducer'
import ReduxFormState from '../../../storybook/Decorators/ReduxFormState'

const initialState = {
  Login: LoginState
}

storiesOf('Login', module)
  .addDecorator(ReduxFormState)
  .add('screen', () => (
    <LoginContainer
      loginState={initialState.Login}
      somethingComplicated={initialState.Login.data.unwrap_or('N/A')}
      login={action('login called')}/>
  ))
