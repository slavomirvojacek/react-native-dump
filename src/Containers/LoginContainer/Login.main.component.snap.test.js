import React from 'react'

import Login from './Login.main.component'
import { props as testDefaultProps } from './defaults'
import { assertSnapshots } from '../../TestUtils/snapshot'

jest
  .mock('../LoginForm/Login.form', () => 'LoginForm')

describe('Login Snapshots', () => {
  const configs = [
    {
      props: {...testDefaultProps},
      desc: 'renders correctly'
    }
  ]

  assertSnapshots(Login, configs)
})
