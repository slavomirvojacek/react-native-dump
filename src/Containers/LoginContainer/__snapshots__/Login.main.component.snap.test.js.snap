// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshots renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <LoginForm
    onSubmit={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    {"data":{},"isPending":false,"error":{}}
  </Text>
</View>
`;
